src/writer/direntPool.h

```diff
@@ -43,9 +43,28 @@ namespace zim
         DirentPool(const DirentPool&) = delete;
         DirentPool& operator=(const DirentPool&) = delete;
         ~DirentPool() {
-          for(auto direntArray: pools) {
-            delete[] direntArray;
+          auto nbPools = pools.size();
+          if (nbPools == 0) {
+            return;
           }
+          // Delete all but last pools (add call the destructors of the dirents)
+          for (auto i = 0U; i<nbPools-1; i++) {
+            auto pool = pools[i];
+            for (auto j = 0U; j < 0xFFFF; j++) {
+              try {
+                pool[j].~Dirent();
+              } catch (...){ /*discard */ }
+            }
```

Comment:
I guess you `try/catch` here in order to respect the C++ requirement that destructors do not throw. But then why don't do it in the `Dirent`'s destructor (and why do you at all think that it can `throw`)?

Reply:
The `try/catch` is here to respect the delete[] behavior (as if we were deleting a array of Dirent)

Here (https://en.cppreference.com/w/cpp/language/delete) it is said :

> If expression is not a null pointer and the deallocation function is not a destroying delete (since C++20), the delete expression invokes the destructor (if any) for the object that's being destroyed, or for every element of the array being destroyed (proceeding from the last element to the first element of the array).
>
> After that, **whether or not an exception was thrown by any destructor**, the delete expression invokes the deallocation function: either operator delete (for the first version of the expression) or operator delete[] (for the second version of the expression), unless the matching new-expression was combined with another new-expression (since C++14). 

So I catch any exception and discard it.